src/audio/module_adapter/module/generic.c

```diff
@@ -347,15 +347,27 @@ int module_free(struct processing_module *mod)
 int module_set_configuration(struct processing_module *mod,
 			     uint32_t config_id,
 			     enum module_cfg_fragment_position pos, size_t data_offset_size,
-			     const uint8_t *fragment, size_t fragment_size, uint8_t *response,
+			     const uint8_t *fragment_in, size_t fragment_size, uint8_t *response,
 			     size_t response_size)
 {
+#if CONFIG_IPC_MAJOR_3
+	struct sof_ipc_ctrl_data *cdata = (struct sof_ipc_ctrl_data *)fragment_in;
+	const uint8_t *fragment = (const uint8_t *)cdata->data[0].data;
+#elif CONFIG_IPC_MAJOR_4
+	const uint8_t *fragment = fragment_in;
+#endif
 	struct module_data *md = &mod->priv;
 	struct comp_dev *dev = mod->dev;
 	size_t offset = 0;
 	uint8_t *dst;
 	int ret;
 
+#if CONFIG_IPC_MAJOR_3
+	if (cdata->cmd != SOF_CTRL_CMD_BINARY) {
```

Comment:
but I'm uncertain about this now. 7d0a0c673f94fe59fbe9b7586f841434ac2569b4 says, that data_blob.c only handles binary controls, that's why that check is added there. But here any control type can be handled?

Reply:
I think you are right. Will remove this.